snforge_std/src/byte_array.cairo

```diff
@@ -1,7 +1,21 @@
+use core::byte_array::BYTE_ARRAY_MAGIC;
+
 fn byte_array_as_felt_array(self: @ByteArray) -> Array<felt252> {
     let mut serialized = array![];
 
     self.serialize(ref serialized);
 
     serialized
 }
+
+fn try_deserialize_bytearray_error(x: Array<felt252>) -> Result<ByteArray, ()> {
```

Comment:
```suggestion
fn try_deserialize_bytearray(x: Span<felt252>) -> Result<ByteArray, ()> {
```

`_error` suffix does not tell us enything.

This function does not use `Array`, but `Span`

Reply:
It does not do `Serde::deserialize<ByteArray>()` - it also verifies if the *error* (emitted with `panic!` or `assert!`) string is also prepended with the magic byte which distinguishes it from other types of errors. That's why i added the `error` suffix, since it's used purely for that - the `ByteArray` deserialization itself does not include the magic byte verification.

Reply:
With the input type i completely agree, missed that

Reply:
> It does not do `Serde::deserialize<ByteArray>()` - it also verifies if the _error_ (emitted with `panic!` or `assert!`) string is also prepended with the magic byte which distinguishes it from other types of errors. That's why i added the `error` suffix, since it's used purely for that - the `ByteArray` deserialization itself does not include the magic byte verification.

Ok, I missed this. Agree here